### DISCUSSION TITLE
Favoritise files in the CWD in the file picker

### DISCUSSION THREAD

**Discussion by poliorcetics** [helix-editor/helix#10596] [General] [answered]
#### Describe your feature request

Example situation:

- A Rust workspace with multiple crates, like this repo
- Enter the folder for one crate, e.g. `helix-lsp`
- Open a file, like `src/lib.rs`
- Press `Space+f` to open the file picker
- The files are randomly ordered

I would like for the files in the current working directory (`helix-lsp`) to come first, especially since `helix` does not have a tree view (and I don't really want one).

Additionally, it would be nice to order the files next to the current one (`helix-lsp/src/lib.rs`) and in subdirectories to be ordered even higher.

So, to clarify, I would like the following order:

1) Files directly next to the opened one and in direct subdirectories that are next to the opened file
2) Files in the current working directory **if if is more precise than the root one** (e.g, inside a specific crate of a cargo workspace)
3) Files in the root directory (e.g. those in the whole cargo workspace)

If you think this is relatively simple to do, I'm ok to work on it to make my first contrib to `helix` :)

**Comment by Triton171** [comment]
I've already implemented the first part in [this PR](https://github.com/helix-editor/helix/pull/1066). We could do the same for files that are next to the currently opened file but in my opinion they shouldn't be prioritized too much.

**Comment by poliorcetics** [comment, marked as answer]
There is now the "directory of current file" picker so I think it's reasonable to mark this as done
